contract_payment_mode/tests/test_contract_payment.py

```diff
@@ -0,0 +1,131 @@
+                    "payment_type": "inbound",
+                }
+            )
+
+        cls.payment_mode = cls.env["account.payment.mode"].create(
+            {
+                "name": "Test payment mode",
+                "active": True,
+                "payment_method_id": cls.payment_method.id,
+                "bank_account_link": "variable",
+            }
+        )
+        cls.partner = cls.env["res.partner"].create(
+            {
+                "name": "Test contract partner",
+                "customer_payment_mode_id": cls.payment_mode.id,
+            }
+        )
+        cls.product = cls.env["product.product"].create(
+            {
+                "name": "Custom Service",
+                "type": "service",
+                "uom_id": cls.env.ref("uom.product_uom_hour").id,
+                "uom_po_id": cls.env.ref("uom.product_uom_hour").id,
+                "sale_ok": True,
+            }
+        )
+        cls.contract = cls.env["contract.contract"].create(
+            {"name": "Maintenance of Servers", "partner_id": cls.partner.id}
+        )
+        company = cls.env.ref("base.main_company")
+        cls.journal = cls.env["account.journal"].create(
+            {
+                "name": "Sale Journal - Test",
+                "code": "HRTSJ",
+                "type": "sale",
+                "company_id": company.id,
+            }
+        )
+
+    def test_post_init_hook(self):
+        contract = self.env["contract.contract"].create(
+            {
+                "name": "Test contract",
+                "partner_id": self.partner.id,
+                "payment_mode_id": self.payment_mode.id,
+            }
+        )
+        self.assertEqual(contract.payment_mode_id, self.payment_mode)
+
+        contract.payment_mode_id = False
+        self.assertEqual(contract.payment_mode_id.id, False)
+
+        post_init_hook(self.env)
+        self.assertEqual(contract.payment_mode_id, self.payment_mode)
+
+    def test_contract_and_invoices(self):
+        self.contract.write({"partner_id": self.partner.id})
+        self.assertEqual(
+            self.contract.payment_mode_id,
+            self.contract.partner_id.customer_payment_mode_id,
+        )
+        self.contract.write(
+            {
+                "line_recurrence": True,
+                "contract_type": "sale",
+                "recurring_interval": 1,
+                "recurring_rule_type": "monthly",
+                "date_start": "2018-01-15",
+                "contract_line_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": self.product.id,
+                            "name": "Database Administration 25",
+                            "quantity": 2.0,
+                            "uom_id": self.product.uom_id.id,
+                            "price_unit": 200.0,
+                        },
+                    )
+                ],
```

Comment:
Please use [Command.create](https://github.com/odoo/odoo/blob/9748b45b24140416fc6e0e87c25c60285bef3d70/odoo/fields.py#L4259); it's more readable.

```suggestion
                "contract_line_ids": [
                    Command.create(
                        {
                            "product_id": self.product.id,
                            "name": "Database Administration 25",
                            "quantity": 2.0,
                            "uom_id": self.product.uom_id.id,
                            "price_unit": 200.0,
                        },
                    )
                ],
```